src/restore.c

```diff
@@ -1168,6 +1169,13 @@ check_index(aerospike *as, index_param *index, uint32_t timeout)
 		}
 	}
 
+	if (!compare_strs(index->ctx, index2.ctx)) {
```

Comment:
With the new check, won't this enter the body of the if statement if both `index->ctx` and `index2.ctx` are NULL (null pointers)? Is that expected?